play-json-tools/src/main/scala/com/evolutiongaming/playjsontools/PlayJsonHelper.scala

```diff
@@ -393,6 +393,57 @@ object PlayJsonHelper {
     }
   }
 
+  object DiscriminatedEitherFormat {
+    def valueFromOnlyKey[A](key: String)(implicit ra: Reads[A]): Reads[A] =
+      new Reads[A] {
+        def reads(js: JsValue) = js match {
+          case o: JsObject => o.keys match {
+            case a if a == Set(key) => ra.reads(o.values.head)
+            case _ => JsError(s"inappropriate set of discriminator keys for either, expected Set(${key}) received ${o}")
+          }
+          case j => JsError(s"can't parse either for non-object for ${j}")
+        }
+      }
+
+    def eitherReads[A, B](
+      keyLeft: String,
+      keyRight: String,
+    )(
+      implicit ra: Reads[A],
+      rb: Reads[B],
+    ): Reads[Either[A, B]] =
+        Reads[Either[A, B]] { json =>
+          valueFromOnlyKey[A](keyLeft).reads(json) match {
```

Comment:
looks overcomplicated, can we just check the presence of "right" key and if not fallback to "left" and not fallback form parsing errors ?